tests/surrogate_test.py

```diff
@@ -156,13 +196,43 @@ def test_train_from_data():
     """Test if train_from_data_works."""
     grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(4))
     surrogate = Surrogate([(-5, 10), (0, 15)], grid_gen)
+    gradient_surrogate = GradientSurrogate([(-5, 10), (0, 15)], grid_gen)
```

Comment:
Again, separate test! And here it is even more important since the `linear_solver` is not a parameter for the `GradientSurrogate`.